python/ray/serve/batching.py

```diff
@@ -250,13 +251,13 @@ async def _assign_func_results(
     async def _process_batches(self, func: Callable) -> None:
         """Loops infinitely and processes queued request batches."""
 
-        while True:
+        while not self._loop.is_closed():
             try:
                 self.curr_iteration_start_time = time.time()
                 await self._process_batch(func)
             except Exception:
                 logger.exception(
-                    "_process_batches asyncio task ran into an " "unexpected exception."
+                    "_process_batches asyncio task ran into an unexpected exception."
                 )
 
```

Comment:
Strangely, it doesn't get logged. I changed the function to:

<details>
  <summary>Modified _process_batches</summary>
  
  ```python
  async def _process_batches(self, func: Callable) -> None:
          """Loops infinitely and processes queued request batches."""
  
          while not self._loop.is_closed():
              logger.warning("Starting a while loop iteration.")
              try:
                  self.curr_iteration_start_time = time.time()
                  await self._process_batch(func)
                  logger.warning("Finished a while loop iteration.")
              except Exception:
                  logger.exception(
                      "_process_batches asyncio task ran into an unexpected exception."
                  )
          
          logger.warning("Do you see me?")
  ```
</details>



I used this app:

<details>
  <summary>Serve app</summary>
  
  ```python
  from ray import serve
  
  @serve.deployment(graceful_shutdown_timeout_s=0.1)
  class GetList:
      def __init__(self):
          self.count = 0
  
      @serve.batch(max_batch_size=2, batch_wait_timeout_s=10)
      async def handle_batch(self, requests):
          return [0] * len(requests)
  
      async def __call__(self, request):
          return await self.handle_batch(request)
  
  app = GetList.bind()
  ```
</details>

I started the app with `serve run try:app`, and then I ran a few requests. Then I ran one request (so the batch wouldn't run), and then killed the app with ctrl-C.

Here's the log file:

[replica_default_GetList_qsc3m3uz.log](https://github.com/ray-project/ray/files/14130350/replica_default_GetList_qsc3m3uz.log)

We don't see either the `Finished a while loop iteration.` or `Do you see me?` logs. This also strange because yesterday when I ran unit tests with a log after the while loop, it did get printed. Maybe the logs only show up if we start Serve using `serve.run()` and the event loop gets killed explicitly (e.g. by `pytest`)?